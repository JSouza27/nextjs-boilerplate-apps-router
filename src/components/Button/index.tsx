import { HtmlHTMLAttributes } from 'react';

type ButtonType = {
  text: string;
} & HtmlHTMLAttributes<HTMLElement>;

export default function Button({ text, ...props }: ButtonType) {
  return (
    <button className="min-w-[4rem] px-4 py-2 rounded- bg-blue-600" {...props}>
      <span>{text}</span>
    </button>
  );
}
