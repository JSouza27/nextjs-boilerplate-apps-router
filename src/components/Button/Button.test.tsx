import { render, screen } from '@testing-library/react';

import Button from '.';

describe('<Button />', () => {
  it('should render the heading', () => {
    const { container } = render(<Button text="Botão" />);

    expect(screen.getByRole('button', { name: 'Botão' })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
