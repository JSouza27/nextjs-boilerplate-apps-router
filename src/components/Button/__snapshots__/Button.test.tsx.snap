// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the heading 1`] = `
<button
  class="min-w-[4rem] px-4 py-2 rounded- bg-blue-600"
>
  <span>
    Bot√£o
  </span>
</button>
`;
